service/api.go

```diff
@@ -440,6 +440,80 @@ func (as *APIServer) SetDownstreamParams(w http.ResponseWriter, r *http.Request)
 	gimlet.WriteJSON(w, fmt.Sprintf("Downstream patches for %v have successfully been set", p.Id))
 }
 
+// NewPush updates when a task is pushing to s3 for s3 copy
+func (as *APIServer) NewPush(w http.ResponseWriter, r *http.Request) {
+	task := MustHaveTask(r)
+	s3CopyReq := &apimodels.S3CopyRequest{}
+
+	if err := utility.ReadJSON(utility.NewRequestReader(r), s3CopyReq); err != nil {
+		as.LoggedError(w, r, http.StatusBadRequest, err)
+		return
+	}
+
+	// Get the version for this task, so we can check if it has
+	// any already-done pushes
+	v, err := model.VersionFindOne(model.VersionById(task.Version))
+	if err != nil {
+		as.LoggedError(w, r, http.StatusInternalServerError,
+			errors.Wrapf(err, "problem querying task %s with version id %s",
+				task.Id, task.Version))
+		return
+	}
+
+	// Check for an already-pushed file with this same file path,
+	// but from a conflicting or newer commit sequence num
+	if v == nil {
+		as.LoggedError(w, r, http.StatusNotFound,
+			errors.Errorf("no version found for build '%s'", task.BuildId))
+		return
+	}
+
+	copyToLocation := strings.Join([]string{s3CopyReq.S3DestinationBucket, s3CopyReq.S3DestinationPath}, "/")
+
+	newestPushLog, err := model.FindPushLogAfter(copyToLocation, v.RevisionOrderNumber)
+	if err != nil {
+		as.LoggedError(w, r, http.StatusInternalServerError,
+			errors.Wrapf(err, "problem querying for push log at %s (build=%s)",
+				copyToLocation, task.BuildId))
+		return
+	}
+	if newestPushLog != nil {
+		grip.Warningln("conflict with existing pushed file:", copyToLocation)
+		gimlet.WriteJSON(w, nil)
+		return
```

Comment:
Are we not returning an error here to avoid the retries? Can we drop a comment here to that effect if so?

Reply:
I don't think that that is the reason. There being a conflict is not an error state. We just don't want to push again so we noop.

Reply:
I think not being able to put something does seem like an error (StatusConflict probably) and that error could be parsed instead of checking TaskId != "", but if that would retry then we wouldn't want to do that.

Reply:
Ah good point. Yes, it probably would retry.